server/src/pages/shotindex/controller.js

```diff
@@ -25,33 +25,55 @@ exports.launch = function(m) {
     }
     Object.assign(m, extractQueryParamValues(queryParamModelPropertyMap));
     model = m;
-    render();
-    if (firstRun && m.shots === null) {
-      // The actual shot data hasn't been loaded yet, so we'll immediately request it:
-      refreshModel();
-    }
-    firstRun = false;
-    return;
   }
+
   if (window.wantsauth) {
-    if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
-      location.search = "reloaded";
+    // Handle non-owner my shots page redirection
+    if (!m.hasDeviceId) {
+      if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
+        location.search = "reloaded";
+        return;
+      }
+      const authTimeout = setTimeout(() => {
+        // eslint-disable-next-line no-global-assign, no-native-reassign
+        location = location.origin + "/#tour";
+      }, FIVE_SECONDS);
+      window.wantsauth.addAuthDataListener((data) => {
+        if (location.search.indexOf("reloaded") > -1) {
+          return;
+        }
+        clearTimeout(authTimeout);
+        location.search = "reloaded";
+      });
       return;
     }
-    const authTimeout = setTimeout(() => {
-      // eslint-disable-next-line no-global-assign, no-native-reassign
-      location = location.origin + "/#tour";
-    }, FIVE_SECONDS);
-    window.wantsauth.addAuthDataListener((data) => {
-      if (location.search.indexOf("reloaded") > -1) {
-        return;
+
+    if (!model.accountId && !model.hasFxa) {
+      if (window.wantsauth.getAuthData()) {
+        Object.assign(model, window.wantsauth.getAuthData());
+        updateModel();
+      } else {
+        window.wantsauth.addAuthDataListener((data) => {
+          Object.assign(model, data);
+          updateModel();
+          render();
+        });
       }
-      clearTimeout(authTimeout);
-      location.search = "reloaded";
-    });
+    }
+  }
+
+  if (firstRun && m.shots === null) {
```

Comment:
...and make this an `else-if` of https://github.com/mozilla-services/screenshots/pull/4759/files#diff-6db5e3f6a06e986e2aa9ab0f5c0170d1R30